src/main/java/org/apache/sling/auth/oauth_client/impl/OidcAuthenticationHandler.java

```diff
@@ -187,32 +198,40 @@ public AuthenticationInfo extractCredentials(@NotNull HttpServletRequest request
             return authInfo;
         }
 
-        //The request is not authenticated. 
-        // 1. Check if the State cookie match with the state in the request received from the idp
+        //The request is not authenticated.
+        // 1. Extract nonce cookie and state cookie from the request
         StringBuffer requestURL = request.getRequestURL();
         if ( request.getQueryString() != null )
             requestURL.append('?').append(request.getQueryString());
 
         Optional<OAuthState> clientState; //state returned by the idp in the redirect request
         String authCode; //authorization code returned by the idp in the redirect request
         Cookie stateCookie;
+        Cookie nonceCookie;
+        Cookie codeVerifierCookie = null;
+        AuthorizationResponse authResponse;
         try {
-            AuthorizationResponse authResponse = AuthorizationResponse.parse(new URI(requestURL.toString()));
+            authResponse = AuthorizationResponse.parse(new URI(requestURL.toString()));
             clientState = extractClientState(authResponse);
-            authCode = extractAuthCode(authResponse);
-            stateCookie = extractStateCookie(request);
-        } catch (ParseException | URISyntaxException e) {
-            logger.debug("Failed to parse authorization response");
+        } catch (ParseException | URISyntaxException | IllegalStateException e) {
+            // If we fail parsing the response, we consider the request not for this authentication handler
```

Comment:
Did you remove the debug logging on purpose? I think it could be useful for troubleshooting